src/wazuh_modules/agent_upgrade/manager/wm_agent_upgrade_tasks.c

```diff
@@ -120,6 +126,16 @@ void wm_agent_upgrade_free_agent_task(wm_agent_task* agent_task) {
     agent_task = NULL;
 }
 
+void wm_agent_upgrade_free_agent_status_task(wm_upgrade_agent_status_task* task) {
+    if (task->message) {
```

Comment:
Check if task is not NULL

Reply:
It is done in the function that calls this. I can add an assertion just in case but that should never be the case